L1Trigger/CSCTriggerPrimitives/src/CSCGEMMotherboard.cc

```diff
@@ -332,3 +359,144 @@ unsigned int CSCGEMMotherboard::findQualityGEM(const CSCALCTDigi& aLCT, const CS
   }
   return quality;
 }
+
+template<>
+void CSCGEMMotherboard::matchingPads<GEMPadDigi>(const CSCALCTDigi& alct,
+						 enum CSCPart part,
+						 matches<GEMPadDigi>& result) const
+{
+  result.clear();
+  if (not alct.isValid()) return;
+
+  std::pair<int,int> alctRoll = (getLUT()->CSCGEMMotherboardLUT::get_csc_wg_to_gem_roll(par))[alct.getKeyWG()];
+  GEMPadDigiIdsBX lut = pads_;
+  for (const auto& p: lut[alct.getBX()]){
+    auto padRoll(getRoll(p));
+    // only pads in overlap are good for ME1A
+    if (part==CSCPart::ME1A and !isPadInOverlap(padRoll)) continue;
+
+    int pad_bx = getBX(p.second)+lct_central_bx;
+    if (std::abs(alct.getBX()-pad_bx)>maxDeltaBXPad_) continue;
+
+    if (alctRoll.first == -99 and alctRoll.second == -99) continue;  //invalid region
+    else if (alctRoll.first == -99 and !(padRoll <= alctRoll.second)) continue; // top of the chamber
+    else if (alctRoll.second == -99 and !(padRoll >= alctRoll.first)) continue; // bottom of the chamber
+    else if ((alctRoll.first != -99 and alctRoll.second != -99) and // center
+	     (alctRoll.first > padRoll or padRoll > alctRoll.second)) continue;
+    result.push_back(p);
+  }
+}
+
+template<>
+void CSCGEMMotherboard::matchingPads<GEMCoPadDigi>(const CSCALCTDigi& alct,
```

Comment:
I tried that already for `pads_` and `copads_`. I had a template function that would access either `pads_` or `copads_` depending on the template argument. But if IIRC it would run-time errors according to David Lange.

Reply:
@davidlange6 do you recall what the problem was here? I don't see how full specialization of a member function template would cause a run-time error.

Reply:
It was a construct like
<PRE>
if (std::is_same<T, GEMPadDigi>::value) temp_lut = pads_;
else  temp_lut = coPads_;
</PRE>

Reply:
@dildick ah, that sort of construct is antithetical to template metaprogramming. You should do this:

```
class CSCGemMotherBoard {
...
template <class P>
const GEMPads& getPads() { return GEMPads(); }
};
template <> const GEMPads& getPads<GEMPadDigi>() { return pads_; }
template <> const GEMPads& getPads<GEMCoPadDigi>() { return coPads_; }
```
Does that make sense?

Reply:
No, the point is that `pads_` and `copads_` also have a different type, so the template suggestion you provided does not work.

Reply:
Ah, I did not realize that. There is probably still a way to do it, but it may be clunkier. Let me think for a bit...

Reply:
Since those variables have types that are also templated on the digi types, you can just do this:
```
class CSCGemMotherBoard {
...
template <class P>
const matchesBX<P>& getPads() const;
};
template <> const matchesBX<GEMPadDigi>& CSCGemMotherBoard::getPads<GEMPadDigi>() const { return pads_; }
template <> const matchesBX<GEMCoPadDigi>& CSCGemMotherBoard::getPads<GEMCoPadDigi>() const { return coPads_; }
```
and then in `matchingPads()`:
```
const matchesBX<T>& lut = getPads<T>();
```